src/Magnum/ImGuiIntegration/Integration.cpp

```diff
@@ -0,0 +1,259 @@
+                   *_instance);
+    return *_instance;
+}
+
+Context::Context() {
+    CORRADE_ASSERT(_instance == nullptr,
+            "ImGuiIntegration::Context::Context(): Context already created.", );
+    ImGui::CreateContext();
+
+    ImGuiIO &io = ImGui::GetIO();
+    io.KeyMap[ImGuiKey_Tab]        = ImGuiKey_Tab;
+    io.KeyMap[ImGuiKey_LeftArrow]  = ImGuiKey_LeftArrow;
+    io.KeyMap[ImGuiKey_RightArrow] = ImGuiKey_RightArrow;
+    io.KeyMap[ImGuiKey_UpArrow]    = ImGuiKey_UpArrow;
+    io.KeyMap[ImGuiKey_DownArrow]  = ImGuiKey_DownArrow;
+    io.KeyMap[ImGuiKey_PageUp]     = ImGuiKey_PageUp;
+    io.KeyMap[ImGuiKey_PageDown]   = ImGuiKey_PageDown;
+    io.KeyMap[ImGuiKey_Home]       = ImGuiKey_Home;
+    io.KeyMap[ImGuiKey_End]        = ImGuiKey_End;
+    io.KeyMap[ImGuiKey_Delete]     = ImGuiKey_Delete;
+    io.KeyMap[ImGuiKey_Backspace]  = ImGuiKey_Backspace;
+    io.KeyMap[ImGuiKey_Enter]      = ImGuiKey_Enter;
+    io.KeyMap[ImGuiKey_Escape]     = ImGuiKey_Escape;
+    io.KeyMap[ImGuiKey_A]          = ImGuiKey_A;
+    io.KeyMap[ImGuiKey_C]          = ImGuiKey_C;
+    io.KeyMap[ImGuiKey_V]          = ImGuiKey_V;
+    io.KeyMap[ImGuiKey_X]          = ImGuiKey_X;
+    io.KeyMap[ImGuiKey_Y]          = ImGuiKey_Y;
+    io.KeyMap[ImGuiKey_Z]          = ImGuiKey_Z;
+
+    /* TODO: Set clipboard text once Platform supports it */
+
+    unsigned char *pixels;
+    int width, height;
+    int pixelSize;
+    io.Fonts->GetTexDataAsRGBA32(&pixels, &width, &height, &pixelSize);
+
+    CORRADE_ASSERT(width > 0,
+            "ImGuiIntegration::Context::Context(): Expected width to be larger than 0", );
+    CORRADE_ASSERT(height > 0,
+            "ImGuiIntegration::Context::Context(): Expected height to be larger than 0", );
+    CORRADE_ASSERT(pixelSize > 0,
+            "ImGuiIntegration::Context::Context(): Expected pixel size to be larger than 0", );
+
+    ImageView2D image{GL::PixelFormat::RGBA,
+        GL::PixelType::UnsignedByte, {width, height},
+        {pixels, std::size_t(pixelSize*width*height)}};
+
+    _texture.setMagnificationFilter(GL::SamplerFilter::Linear)
+        .setMinificationFilter(GL::SamplerFilter::Linear)
+        .setStorage(1, GL::TextureFormat::RGBA8, image.size())
+        .setSubImage(0, {}, image);
+
+    _mesh.setPrimitive(GL::MeshPrimitive::Triangles);
+    _mesh.addVertexBuffer(
+        _vertexBuffer, 0, Implementation::ImGuiShader::Position{},
+        Implementation::ImGuiShader::TextureCoordinates{},
+        Implementation::ImGuiShader::Color(Implementation::ImGuiShader::Color::Components::Four,
+            Implementation::ImGuiShader::Color::DataType::UnsignedByte,
+            Implementation::ImGuiShader::Color::DataOption::Normalized));
+
+    _timeline.start();
+
+    _instance = this;
+}
+
+Context::~Context() {
+    /* Not initialized/initialization failed */
+    if(!_instance) return;
+
+    ImGui::DestroyContext();
+
+    _instance = nullptr;
```

Comment:
`CORRADE_INTERNAL_ASSERT(_instance == this);` before to prevent accidental bugs later